## Data and Code from the Microbial Ecology Study of an Arctic Travertine-forming Geothermal Spring

This paper can be accessed at https://www.liebertpub.com/doi/10.1089/ast.2023.0130

Cite as:

>Ugwuanyi IR, Steele A, Glamoclija M. Microbial Ecology of an Arctic Travertine Geothermal Spring: Implications for Biosignature Preservation and Astrobiology. Astrobiology 2024;24(7):734-753; doi: 10.1089/ast.2023.0130


This repository contains the link to metagenome sequences, data, and the code used to investigate the microbial ecology of a 
travertine-forming geothermal spring in the Arctic and its astrobiological relevance.

Processing of metagenome sequence was performed using a suite of freely available software installed on 
[Rutgers Amarel Cluster](https://oarc.rutgers.edu/resources/amarel/). Data analyses were performed
using [R software](https://www.r-project.org/).

The Scripts directory contains a list of software and all the scripts used to process 
raw metagenome sequences and build metagenome-assembled genomes (MAGs). The Data_files directory contains the 
data files generated after processing metagenome sequences. The Data_files directory contains the taxonomy table,
MAG taxonomy table, MAG abundance tables, and all files used as input for data analyses in R. The file JS_metagenome_sequence_analysis.r contains the 
R code used to analyze data generated after processing metagenome sequences. The raw metagenome sequences and MAGs 
can be found in the NCBI SRA archive under Bioproject number [PRJNA1080596](https://www.ncbi.nlm.nih.gov/bioproject/?term=PRJNA1080596).

For questions, please contact me. 
